Fixed keybinds activating when typing in the debug console

ZehsTeam opened the pull request:
- Fixed keybinds activating when typing in the debug console.
- Fixed ToggleHUD keybind activating when typing in the imperium interface, chat, and debug console.
- Added `.vs/` to gitignore.

ZehsTeam committed:
Fixed keybinds activating when typing in the debug console

1A3Dev commented on Imperium/src/Interface/ImperiumUI/Windows/Teleport/Widgets/Teleportation.cs:
```diff
@@ -181,7 +181,7 @@ private void SetExtractionPoints()
     private static void OnInteractiveTeleport(InputAction.CallbackContext callbackContext)
     {
-        if (Imperium.Interface.IsOpen() || MenuManager.instance.IsOpen() || ChatManager.instance.IsOpen()) return;
+        if (Imperium.Interface.IsOpen() || MenuManager.instance.IsOpen() || ChatManager.instance.IsOpen() || DebugConsoleUI.instance.IsOpen()) return;
```
```suggestion
        if (Imperium.Interface.IsOpen() || !SemiFunc.NoTextInputsActive()) return;
```

1A3Dev commented on Imperium/src/Core/Scripts/TapeMeasure.cs:
```diff
@@ -148,7 +148,7 @@ private void OnTapeOpen(InputAction.CallbackContext callbackContext)
             OnExitAction();
         }
-        else if (!Imperium.Interface.IsOpen() && !MenuManager.instance.IsOpen() && !ChatManager.instance.IsOpen())
+        else if (!Imperium.Interface.IsOpen() && !MenuManager.instance.IsOpen() && !ChatManager.instance.IsOpen() && !DebugConsoleUI.instance.IsOpen())
```
```suggestion
        else if (!Imperium.Interface.IsOpen() && SemiFunc.NoTextInputsActive())
```

1A3Dev commented on Imperium/src/Interface/MapUI/MinimapOverlay.cs:
```diff
@@ -117,6 +117,7 @@ private void Update()
             || Imperium.Interface.IsOpen()
             || MenuManager.instance.IsOpen()
             || ChatManager.instance.IsOpen()
+            || DebugConsoleUI.instance.IsOpen()
```
```suggestion
|| !SemiFunc.NoTextInputsActive()
```

1A3Dev commented on Imperium/src/Core/Scripts/ImpFreecam.cs:
```diff
@@ -181,7 +182,7 @@ private void OnFreecamDisable()
     private void OnFreecamReset(InputAction.CallbackContext callbackContext)
     {
-        if (Imperium.Interface.IsOpen() || MenuManager.instance.IsOpen() || ChatManager.instance.IsOpen()) return;
+        if (Imperium.Interface.IsOpen() || MenuManager.instance.IsOpen() || ChatManager.instance.IsOpen() || DebugConsoleUI.instance.IsOpen()) return;
```
```suggestion
        if (Imperium.Interface.IsOpen() || !SemiFunc.NoTextInputsActive()) return;
```

1A3Dev commented on Imperium/src/Core/Scripts/ImpFreecam.cs:
```diff
@@ -111,6 +111,7 @@ private void OnMinicamFullscreenToggle(InputAction.CallbackContext callbackConte
         if (Imperium.Interface.IsOpen() ||
             MenuManager.instance.IsOpen() ||
             ChatManager.instance.IsOpen() ||
+            DebugConsoleUI.instance.IsOpen() ||
```
```suggestion
 !SemiFunc.NoTextInputsActive() ||
```

1A3Dev commented on Imperium/src/Core/Scripts/ImpFreecam.cs:
```diff
@@ -211,7 +212,7 @@ private static void OnFreehandModeDisable()
     private void OnToggleLayerSelector(InputAction.CallbackContext callbackContext)
     {
-        if (Imperium.Interface.IsOpen() || MenuManager.instance.IsOpen() || ChatManager.instance.IsOpen()) return;
+        if (Imperium.Interface.IsOpen() || MenuManager.instance.IsOpen() || ChatManager.instance.IsOpen() || DebugConsoleUI.instance.IsOpen()) return;
```
```suggestion
        if (Imperium.Interface.IsOpen() || !SemiFunc.NoTextInputsActive()) return;
```

1A3Dev commented on Imperium/src/Core/Lifecycle/WaypointManager.cs:
```diff
@@ -109,6 +110,7 @@ private void OnWaypointTeleport(InputAction.CallbackContext _)
         if (Imperium.Interface.IsOpen() ||
             MenuManager.instance.IsOpen() ||
             ChatManager.instance.IsOpen() ||
+            DebugConsoleUI.instance.IsOpen() ||
```
```suggestion
!SemiFunc.NoTextInputsActive() ||
```

1A3Dev commented on Imperium/src/Core/Lifecycle/WaypointManager.cs:
```diff
@@ -93,6 +93,7 @@ private void OnWaypointDelete(InputAction.CallbackContext _)
         if (Imperium.Interface.IsOpen() ||
             MenuManager.instance.IsOpen() ||
             ChatManager.instance.IsOpen() ||
+            DebugConsoleUI.instance.IsOpen() ||
```
```suggestion
!SemiFunc.NoTextInputsActive() ||
```

1A3Dev commented on Imperium/src/Core/Lifecycle/PlayerManager.cs:
```diff
@@ -104,6 +105,8 @@ internal static void ToggleLocalAvatar(bool isShown)
 
     private static void ToggleHUD(InputAction.CallbackContext callbackContext)
     {
+        if (Imperium.Interface.IsOpen() || ChatManager.instance.IsOpen() || DebugConsoleUI.instance.IsOpen()) return;
```
```suggestion
        if (Imperium.Interface.IsOpen() || !SemiFunc.NoTextInputsActive()) return;
```

1A3Dev commented on Imperium/src/Interface/MapUI/MapUI.cs:
```diff
@@ -154,7 +154,7 @@ protected override void OnThemePrimaryUpdate(ImpTheme themeUpdate)
     private static void OnMinimapToggle(InputAction.CallbackContext _)
     {
-        if (Imperium.Interface.IsOpen() || MenuManager.instance.IsOpen() || ChatManager.instance.IsOpen()) return;
+        if (Imperium.Interface.IsOpen() || MenuManager.instance.IsOpen() || ChatManager.instance.IsOpen() || DebugConsoleUI.instance.IsOpen()) return;
```
```suggestion
        if (Imperium.Interface.IsOpen() || !SemiFunc.NoTextInputsActive()) return;
```

1A3Dev commented on Imperium/src/Extensions/DebugConsoleUIExtensions.cs:
This file would no longer be needed with the changes I suggested

ZehsTeam committed:
Apply suggestions from code review

Simplified detecting when a text input is active using `SemiFunc.NoTextInputsActive()`

Co-authored-by: Jenson R <mcplayer1a3@gmail.com>

ZehsTeam committed:
Simplified detecting if an input field is active in more places and some minor formatting changes

ZehsTeam committed:
Removed unused class extension (DebugConsoleUIExtensions.cs)
